src/meta/meta_ckpt.c

```diff
@@ -590,6 +590,10 @@ __wt_meta_ckptlist_set(
     WT_DECL_RET;
     bool has_lsn;
 
+    /* Review the checkpoint's write generation. */
+    WT_CKPT_FOREACH (ckptbase, ckpt)
+        __ckptlist_review_write_gen(session, ckpt);
+
```

Comment:
I believe the current code (before your change), and this version, both update the write generation in all checkpoints, not just the new one. I don't think the write generation is used for anything at the moment and it isn't a problem, but it seems wrong to rewrite historic checkpoints with new write generations.

If you agree, maybe we should move the call to `__ckptlist_review_write_gen()` into `__wt_meta_ckptlist_to_meta()`, and only call it if `WT_CKPT_ADD` is set?

Reply:
0f91e4c